documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -261,4 +263,302 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
+
+**Description**: Verifies whether specification supports HTTPS scheme or not.
+
+**Why the rule is important**: All the ARM specification should support HTTPS endpoint.
+
+**How to fix the violation**: N/A.
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2004" />R2004 NonApplicationJsonType
+**Output Message**: Please make sure that media types other than 'application/json' are supported by your service.
+
+**Description**: Verifies whether operation supports "application/json" as consumes or produces section.
+
+**Why the rule is important**: N/A.
+
+**How to fix the violation**: N/A.
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2004" />R2004 NonApplicationJsonType
+**Output Message**: Please make sure that media types other than 'application/json' are supported by your service.
+
+**Description**: Verifies whether operation supports "application/json" as consumes or produces section.
+
+**Why the rule is important**: N/A.
+
+**How to fix the violation**: N/A.
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2059" />R2059 UniqueResourcePaths
+**Output Message**: Multiple resource providers are not allowed in a single spec. More than one the resource paths were found: '{0}'.
+
+**Description**: Verifies whether more than one resource providers exists in the specification or not.
+
+**Why the rule is important**: Per the ARM guidelines, each swagger specification must contain one resource provider.
+
+**How to fix the violation**: N/A.
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2059" />R2059 UniqueResourcePaths
+**Output Message**: Multiple resource providers are not allowed in a single spec. More than one the resource paths were found: '{0}'.
+
+**Description**: Verifies whether more than one resource providers exists in the specification or not.
```

Comment:
I was thinking more on the lines on a couple of paths with different resource names that should cause a violation.